src/Actions/ResizeImages.php

```diff
@@ -5,26 +5,32 @@
 use JustBetter\ImageOptimize\Jobs\ResizeImageJob;
 use Statamic\Assets\Asset;
 use Statamic\Assets\AssetCollection;
+use Symfony\Component\Console\Helper\ProgressBar;
 
 class ResizeImages
 {
     public int $chunkSize = 200;
 
     public function __construct(
-        AssetCollection $assetCollection
+        AssetCollection $assetCollection,
+        public ?ProgressBar $progressBar = null
     ) {
         $this->resizeImages($assetCollection);
     }
 
     public function resizeImages(AssetCollection $assetCollection): void
     {
-        $assetCollection->chunk($this->chunkSize)->each(function (AssetCollection $assets) {
-            $assets->each(function (Asset $asset) {
-                if($asset->isImage()) {
-                    $asset->hydrate();
+        $assetCollection->lazy()->each(function (Asset $asset) {
+            if($asset->isImage()) {
+                $asset->hydrate();
+
+                if ($this->progressBar) {
+                    ResizeImageJob::dispatchSync($asset);
```

Comment:
This may slow down execution of resizing as it removes the concurrency of `dispatch`

Reply:
This was done on purpose, if you're running through cli with verbose you want to show the real progress. Because if you're running through redis you would get your progress instantly while the actual job probably isn't even done yet.